draft-ietf-quic-version-negotiation.md

```diff
@@ -178,27 +193,32 @@ endpoint might be aware of the compatibility document while the other may not.
 
 ## Compatible Version Negotiation {#compat-vn}
 
-When the server can parse the client's first flight using the original version,
-it can extract the client's Version Information structure (see {{vers-info}}).
-This contains the list of versions that the client knows its first flight is
-compatible with.
-
-If the server supports one of the client's compatible versions, and the server
-also knows that the original version is compatible with this version, and the
-client's first flight is compatible with this version, then the server converts
-the client's first flight to that version and replies to the client as if it had
-received the converted first flight. Note that this conversion process cannot
-fail by definition of the first flight being compatible. The version used by the
-server in its reply is refered to as the "negotiated version". The server MUST
-NOT reply with a version that is not present in the client's compatible
-versions, unless it is the original version.
-
-Clients will be made aware of compatible version negotiation by seeing a change
-in the QUIC long header Version field. It is possible for the server to
-initially send packets with the original version before switching to the
+When the server can parse the client's first flight using the client's
+chosen version, it can extract the client's Version Information structure
+(see {{vers-info}}). This contains the list of versions that the
+client knows its first flight is compatible with.
+
+In order to do compatible version negotiation, the server MUST select
+one of these versions that (1) it supports and (2) it knows is
+compatible with the client's chosen version. Once the server has
```

Comment:
this is backwards. the client's chosen version needs to be compatible with the selected version not the other way around